### PR TITLE
Added interfaces and traits

### PR DESCRIPTION
**Pull request by Tahmid11**
I have added in the trait and interface folders and some of the files.

### CODE REVIEW

File: src/Interfaces/Actioned.php [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+use Illuminate\Contracts\View\View;
+use Illuminate\Database\Eloquent\Relations\MorphMany;
+
+interface Actioned extends ActivityLog
+{
+    public function activities(): MorphMany;
+
+    public function viewActivities(): View;
+}
```
Comment (calum-gray) [inline src/Interfaces/Actioned.php:14]: blank line at end of file

File: src/Traits/HasActions.php [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        ) ->showSubject();
+
+        return GovukPage::custom("Activities performed by {$this->label()}", 'activity', [])
+            ->setBack($this->backRoute())
+            ->with('activities', $actions->toArray(request()))
+            ->with('pagination', $actions->resource->toArray())
+            ->with('showSubject', true)
+            ->with('subject', $this);
+    }
+}
```
Comment (calum-gray) [inline src/Traits/HasActions.php]: blank line at end of file

File: src/Traits/HasActivities.php [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        );
+
+        return GovukPage::custom("Activity log of {$this->label()}", 'activity', [])
+            ->setBack($this->backRoute())
+            ->with('activities', $activities->toArray(request()))
+            ->with('pagination', $activities->resource->toArray())
+            ->with('showSubject', false)
+            ->with('subject', $this);
+    }
+}
```
Comment (calum-gray) [inline src/Traits/HasActivities.php]: blank line at end of file

**Review by calum-gray — CHANGES_REQUESTED** [review]
could you add the activitylog interface too pls

**Review by calum-gray — CHANGES_REQUESTED** [review]
need to add the functions into user.php, see the example branch

**Comment by Tahmid11** [comment]
> need to add the functions into user.php, see the example branch

Thank you for catching that!

File: src/Traits/HasActions.php [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+trait HasActions
+{
+    public function viewActions(): View
+    {
+        $actions = ActivityCollection::make(
+            $this
+                ->actions()
+                ->with(['causer', 'subject'])
+                ->paginate(10),
+        ) ->showSubject();
```
Comment (AnthonyEdmonds) [inline src/Traits/HasActions.php:22]: No space between function brace and arrow

File: tests/Models/User.php [hunk 5]
```diff
@@ -27,4 +33,19 @@ protected static function newFactory(): UserFactory
     {
         return new UserFactory();
     }
+
+    public function backRoute(): string
+    {
+        return route ('users.index', $this);
```
Comment (AnthonyEdmonds) [inline tests/Models/User.php:39]: No space between function name and opening brace.

**Review by AnthonyEdmonds — COMMENTED** [review]
I hope you understand all of this, as you'll have to write tests for it soon!

**Review by calum-gray — APPROVED** [review]

**Pull request merged**